modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractScalaCodegen.java

```diff
@@ -57,6 +58,9 @@ public void processOpts() {
         if (additionalProperties.containsKey(CodegenConstants.SOURCE_FOLDER)) {
             this.setSourceFolder((String) additionalProperties.get(CodegenConstants.SOURCE_FOLDER));
         }
+        if (additionalProperties.get("stripPackageName") != null) {
```

Comment:
"stripPackageName" should be added as constant in CodegenConstants

Use additionalProperties.containsKey() to checkout if the property exists in the map.
I would do it this way
`
if (additionalProperties.containsKey(CodegenConstants.STRIP_PACKAGE_NAME) &&
 "true".equalsIgnoreCase(additionalProperties.get(CodegenConstants.STRIP_PACKAGE_NAME)) {
this. stripPackageName = true;
}
`

Reply:
Regarding `CodegenConstants`, this is currently a scala-only feature, as far as I'm aware. Should language specific options be put there?

Reply:
Regarding the conditional, I would agree if the default for `stripPackageName` was false, but it is true (as noted above), so that code snippet won't work.  But I'm happy to change the code as long as the logic is correct.

Reply:
The plan is to migrate all scala generators into one. 

so the code snippet could be like this.
`
if (additionalProperties.containsKey(CodegenConstants.STRIP_PACKAGE_NAME) && "false".equalsIgnoreCase(additionalProperties.get(CodegenConstants.STRIP_PACKAGE_NAME)) { this. stripPackageName = false; }
`